app/middleware/utils/applyColorsToData.js

```diff
@@ -8,22 +8,8 @@ export default function applyColorsToData(colors = [], data = []) {
     return data;
   }
 
-  function _getColorForIndex(seriesIndex) {
-    let color;
-    let iteration = 0;
-    while (!color) {
-      const colorsIndex = seriesIndex - (iteration * colors.length);
-      if (colorsIndex < colors.length) {
-        color = colors[colorsIndex];
-      } else {
-        iteration++;
-      }
-    }
-    return color;
-  }
-
   return data.map((series, idx) => {
-    series.color = _getColorForIndex(idx); // eslint-disable-line no-param-reassign
+    series.color = loopArrayItemAtIndex(idx, colors); // eslint-disable-line no-param-reassign
```

Comment:
Instead of this disable, do:

You might have to enable the bable stage-2 plugin for this, but it's worth it (it's a shortcut to `Object.assign({}, series, { color })`)

If you are modifying and object you should always clone it (and break the refer) otherwise you get into phantom bug territory.

```
return data.map((series, idx) => ({ ...series, color:loopArrayItemAtIndex(idx, colors) });
```